test/widgets/content_test.dart

```diff
@@ -230,6 +230,49 @@ void main() {
     });
   });
 
+  group('ListNodeWidget', () {
+    testWidgets('ordered list with custom start', (tester) async {
+      await prepareContent(tester, plainContent('<ol start="3">\n<li>third</li>\n<li>fourth</li>\n</ol>'));
+
+      expect(find.text('3. '), findsOneWidget);
+      expect(find.text('4. '), findsOneWidget);
+      expect(find.text('third'), findsOneWidget);
+      expect(find.text('fourth'), findsOneWidget);
+    });
+
+    testWidgets('list uses correct text baseline alignment', (tester) async {
+      await prepareContent(tester, plainContent(ContentExample.orderedListLargeStart.html));
+
+      final table = tester.widget<Table>(find.byType(Table));
+      check(table.defaultVerticalAlignment).equals(TableCellVerticalAlignment.baseline);
+      check(table.textBaseline).equals(localizedTextBaseline(tester.element(find.byType(Table))));
+    });
+
+    testWidgets('long ordered list markers render completely and are right-aligned', (tester) async {
+      await prepareContent(tester, plainContent(ContentExample.orderedListLargeStart.html));
+
+      final markerTexts = tester.widgetList<Text>(
+        find.descendant(of: find.byType(Align), matching: find.byType(Text)));
+        
+      for (final text in markerTexts) {
+        final renderParagraph = tester.renderObject(find.text(text.data!)) as RenderParagraph;
+        final textHeight = renderParagraph.size.height;
+        final lineHeight = renderParagraph.text.style!.height! * renderParagraph.text.style!.fontSize!;
+        check(textHeight).equals(lineHeight);
+
+        final textWidth = renderParagraph.size.width;
+        final renderBox = tester.renderObject(find.ancestor(
+            of: find.text(text.data!),
+            matching: find.byType(Align))) as RenderBox;
+        check(renderBox.size.width >= textWidth).isTrue();
+
+        final align = tester.widget<Align>(
+          find.ancestor(of: find.text(text.data!), matching: find.byType(Align)));
+        check(align.alignment).equals(AlignmentDirectional.topEnd);
```

Comment:
Let's adapt this so that it's in terms of what the user sees, rather than how the implementation code works. (Per https://zulip.readthedocs.io/en/latest/testing/philosophy.html#integration-testing-or-unit-testing again.)

Concretely: the content in this test has two markers. They aren't going to be the same width. So if the alignment is correct, they'll have their right edges at the same x-coordinate, and if the alignment is wrong they'll almost surely have their right edges at different x-coordinates. So to write a test, get the locations of the two marker `Text`s, confirm the widths are different, and then check the right edges match.